### PR TITLE
increase visibility of warnings from getMadratGraph

### PR DESCRIPTION
**Pull request by tscheypidi**
with this update warnings from getMadratGraph will show up in the log files as well as on standard output.

### CODE REVIEW

File: R/getCode.R [hunk 1]
```diff
@@ -1,97 +1,110 @@
...
+      x <- gsub(" +", " ", x)
+      x <- sub(", ?returnPathOnly ?= ?(FALSE|TRUE)", "", x)
       x <- sub("\\)$", ", returnPathOnly = TRUE)", x)
       return(unique(.evals(unique(x))))
     }
-    getMappings <- lapply(getMappings,.getPaths)
+    getMappings <- lapply(getMappings, .getPaths)
+    .warnNotfound <- function(getMappings) {
+      .notfound <- function(set) return(is.element("NOTFOUND", set))
+      notfound <- vapply(getMappings, .notfound, logical(1))
```
Comment (pfuehrlich-pik) [inline R/getCode.R:70]:
```suggestion
      notfound <- vapply(getMappings, function(set) is.element("NOTFOUND", set), logical(1))
```
I'd just inline the function here. In any case I'd omit the explicit `return`.

Comment (pfuehrlich-pik) [reply R/getCode.R:70]: fun fact: in R 4.1 you could even use the new short function syntax
```suggestion
      notfound <- vapply(getMappings, \(set) is.element("NOTFOUND", set), logical(1))
```

File: R/getMadratGraph.R [hunk 2]
```diff
@@ -1,102 +1,103 @@
...
 #' @export
 
-getMadratGraph <- function(packages=installedMadratUniverse(), globalenv=getConfig("globalenv")) {
-  
-  if (is.null(getOption("MadratCache"))) options(MadratCache = new.env(size = NA))
-  
+
+getMadratGraph <- function(packages = installedMadratUniverse(), globalenv = getConfig("globalenv")) {
+
+  if (is.null(getOption("MadratCache"))) options(MadratCache = new.env(size = NA)) # nolint
```
Comment (pfuehrlich-pik) [inline R/getMadratGraph.R:20]: Just pointing out how this would look with `withr`:
```suggestion
  if (is.null(getOption("MadratCache"))) withr::local_options(c(MadratCache = new.env(size = NA)))
```
This will of course only change the option until the end of a getMadratGraph call. If we want to set it globally for the entire session we should leave it as it is.

File: R/retrieveData.R [hunk 3]
```diff
@@ -170,9 +174,9 @@ retrieveData <- function(model, rev = 0, dev = "", cachetype = "rev", ...) {
     vcat(2, " - function ", functionname, " finished", fill = 300, show_prefix = FALSE)
 
     cwd <- getwd()
-    setwd(sourcefolder)
+    setwd(sourcefolder) # nolint
     system(paste0("tar --create --gzip --file ../", collectionname, ".tgz", " ./*"))
```
Comment (pfuehrlich-pik) [inline R/retrieveData.R:178]: If we do this we don't need the `cwd` and `setwd` stuff:
```suggestion
    withr::with_dir(sourcefolder, system(paste0("tar --create --gzip --file ../", collectionname, ".tgz", " ./*")))
```

File: tests/testthat/test-calcOutput.R [hunk 4]
```diff
@@ -16,159 +16,161 @@ nc <- function(x) {
...
                                     class      = "list",
                                     unit       = "1",
                                     description = "test"))
-  globalassign(paste0("calcBla",1))
-  data <- calcOutput("Bla1", aggregate=FALSE, file = "test.rds")
+  globalassign(paste0("calcBla", 1))
+  data <- calcOutput("Bla1", aggregate = FALSE, file = "test.rds")
   expect_equivalent(data, list(1))
-  expect_identical(readRDS(paste0(getConfig("outputfolder"),"/test.rds")),data)
+  expect_identical(readRDS(paste0(getConfig("outputfolder"), "/test.rds")), data)
```
Comment (pfuehrlich-pik) [inline tests/testthat/test-calcOutput.R:173]:
```suggestion
  expect_identical(readRDS(file.path(getConfig("outputfolder"), "test.rds")), data)
```

File: tests/testthat/test-calcOutput.R [hunk 5]
```diff
@@ -231,34 +233,35 @@ test_that("Aggregation works", {
...
+  expect_identical(nc(calcOutput("AggregationTest", aggregate = "regglo")), mbind(reg, glo))
+  expect_warning(a <- nc(calcOutput("AggregationTest", aggregate = "global+region+cheese")),
+                 "Omitting cheese from aggregate")
+  expect_identical(a, mbind(glo, reg))
   expect_error(calcOutput("MalformedAggregation"), "must be a function")
   expect_error(calcOutput("MalformedAggregation2"), "must be a list of function arguments")
-  
-  xtramap <- paste0(tempdir(),"/blub.csv")
+
+  xtramap <- paste0(tempdir(), "/blub.csv")
```
Comment (pfuehrlich-pik) [inline tests/testthat/test-calcOutput.R:260]:
```suggestion
  xtramap <- file.path(tempdir(), "blub.csv")
```

File: tests/testthat/test-calcOutput.R [hunk 6]
```diff
@@ -268,82 +271,83 @@ test_that("Edge cases work as expected", {
...
-  
+  expect_error(calcOutput("NoMag2", aggregate = FALSE), "Weights are currently not supported")
+  expect_error(calcOutput("NoMag3", aggregate = FALSE), "Min/Max checks cannot be used")
+  expect_error(calcOutput("NoMag4", aggregate = FALSE), "Structure checks cannot be used")
+  expect_error(calcOutput("NoMag5", aggregate = FALSE), "isocountries can only be set if")
+
   skip_if_not_installed("reshape2")
   a <- calcOutput("EdgeTest2", file = "blub.mif")
-  b <- magclass::read.report(paste0(getConfig("outputfolder"),"/blub.mif"), as.list = FALSE)
+  b <- magclass::read.report(paste0(getConfig("outputfolder"), "/blub.mif"), as.list = FALSE)
```
Comment (pfuehrlich-pik) [inline tests/testthat/test-calcOutput.R:350]:
```suggestion
  b <- magclass::read.report(file.path(getConfig("outputfolder"), "blub.mif"), as.list = FALSE)
```

File: tests/testthat/test-getMadratInfo.R [hunk 7]
```diff
@@ -2,48 +2,60 @@ context("MADRaT analysis tools")
...
-  expect_message(a <- getMadratInfo(packages="madrat", cutoff = -1, extended = TRUE),"passed")
+  setConfig(globalenv = FALSE, .verbose = FALSE)
+  expect_message(a <- getMadratInfo(packages = "madrat", cutoff = -1, extended = TRUE), "passed")
 })
 
 test_that("getMadratInfo properly detects problems", {
   setConfig(globalenv = TRUE, .verbose = FALSE)
-  calcBla <- function() {type<-"TauTotal";calcOutput(type)}
+  calcBla <- function() {
+type <- "TauTotal"
```
Comment (pfuehrlich-pik) [inline tests/testthat/test-getMadratInfo.R:19]: Indentation seems off

**Review by pfuehrlich-pik — APPROVED** [review]
Seems good. I'd like to introduce `withr` to make the code a little nicer and some #nolint statements unnecessary.

File: R/getMadratGraph.R (hunk 2 above)
Comment (tscheypidi) [reply R/getMadratGraph.R:20]: must be set globally as it does not make sense to create a cache which is immediately deleted again afterwards

File: R/getCode.R [hunk 8]
```diff
@@ -1,97 +1,110 @@
...
 #' @seealso \code{\link{getMadratGraph}}
 
-getCode <- function(packages=installedMadratUniverse(), globalenv=getConfig("globalenv")) {
+getCode <- function(packages = installedMadratUniverse(), globalenv = getConfig("globalenv")) {
   # LC_CTYPE changes how \uFC is deparsed (ü vs <U+00FC>), but we want locale independent results
   ctypeLocale <- Sys.getlocale("LC_CTYPE")
-  Sys.setlocale("LC_CTYPE", "C")
-  on.exit(Sys.setlocale("LC_CTYPE", ctypeLocale))
-  
+  Sys.setlocale("LC_CTYPE", "C") # nolint
```
[the remark this answers is not in the document]
Comment (tscheypidi) [reply R/getCode.R:15]: agreed

File: R/getCode.R (hunk 1 above)
Comment (tscheypidi) [reply R/getCode.R:70]: not sure what the advantage would be

File: R/retrieveData.R (hunk 3 above)
Comment (tscheypidi) [reply R/retrieveData.R:178]: done

File: tests/testthat/test-calcOutput.R (hunk 4 above)
Comment (tscheypidi) [reply tests/testthat/test-calcOutput.R:173]: done

File: tests/testthat/test-calcOutput.R (hunk 5 above)
Comment (tscheypidi) [reply tests/testthat/test-calcOutput.R:260]: done

File: tests/testthat/test-calcOutput.R (hunk 6 above)
Comment (tscheypidi) [reply tests/testthat/test-calcOutput.R:350]: done

File: tests/testthat/test-getMadratInfo.R (hunk 7 above)
Comment (tscheypidi) [reply tests/testthat/test-getMadratInfo.R:19]: fixed
